Raise error when json files are parsed and their root value is an array instead of object. (#225)
bids-standard/bids-validator: changelog.d/20250808_135941_rosswilsonblair_raise_error_on_non_object_json.md

<!--
A new scriv changelog fragment.

Uncomment the section that is right (remove the HTML comment wrapper).
For top level release notes, leave all the headers commented out.
-->

<!--
### Added

- A bullet item for the Added category.

-->
### Changed

> - Raise error when json files are parsed and their root value is an array instead of object

Reviewer: ```suggestion
- Raise error when JSON files are parsed and their root value is anything other than an object
```
